rust/uefi/stub/src/main.rs

```diff
@@ -72,7 +74,60 @@ fn main(handle: Handle, mut system_table: SystemTable<Boot>) -> Status {
     let status;
     // A list of dynamically assembled initrds, e.g. credential initrds or system extension
     // initrds.
-    let dynamic_initrds: Vec<Vec<u8>> = Vec::new();
+    let mut dynamic_initrds: Vec<Vec<u8>> = Vec::new();
+
+    {
+        // This is a block for doing filesystem operations once and for all, related to companion
+        // files, nothing can open the LoadedImage protocol here.
+        // Everything must use `filesystem`.
+        let mut companions = Vec::new();
+        let mut filesystem = uefi::fs::FileSystem::new(
+            system_table
+                .boot_services()
+                .get_image_file_system(system_table.boot_services().image_handle())
+                .expect("Failed to open the simple filesystem for this image; netboot?"),
+        );
+        let default_dropin_directory;
```

Comment:
Could maybe be simmplified to:

```rust
let default_dropin_directory =  pe_in_memory.file_path().map(...).transpose()?;
```

This would also remove the panic.

Reply:
Fixed.